clouddriver-ecs/src/test/groovy/com/netflix/spinnaker/clouddriver/ecs/services/ContainerInformationServiceSpec.groovy

```diff
@@ -149,11 +157,62 @@ class ContainerInformationServiceSpec extends Specification {
     retrievedHealthStatus == expectedHealthStatus
   }
 
-  def 'should return correct health status based on last task status'() {
+  def 'should return correct health status based on last task status with no health checks defined'() {
     setup:
     def taskId = 'task-id'
     def serviceName = 'test-service-name'
-    taskCacheClient.get(_) >> new Task(lastStatus: lastStatus)
+    def cachedService = new Service(
+      serviceName: serviceName,
+      loadBalancers: [new LoadBalancer()]
+    )
+
+    serviceCacheClient.get(_) >> cachedService
+    taskCacheClient.get(_) >> new Task(lastStatus: lastStatus, healthStatus: healthStatus)
+    taskDefinitionCacheClient.get(_) >> new TaskDefinition()
+    def expectedHealthStatus = [
+      [
+        instanceId: taskId,
+        state     : 'Unknown',
+        type      : 'loadBalancer'
+      ],
+      [
+        instanceId: taskId,
+        state     : resultStatus,
+        type      : 'ecs',
+        healthClass: 'platform'
+      ]
+    ]
+    def retrievedHealthStatus = service.getHealthStatus(taskId, serviceName, 'test-account', 'us-west-1')
+
+    expect:
+    retrievedHealthStatus == expectedHealthStatus
+
+    where:
+    healthStatus  | resultStatus  | lastStatus
+    'UNKNOWN'     | 'Starting'    | 'PROVISIONING'
+    'UNKNOWN'     | 'Starting'    | 'PENDING'
+    'UNKNOWN'     | 'Starting'    | 'ACTIVATING'
+    'UNKNOWN'     | 'Unknown'     | 'RUNNING'
```

Comment:
is this test case saying that a "running" task with "unknown" healthStatus and no health-check is expected to be "unknown" ? would to be expected to be "up" since healthStatus will always be unknown?

Reply:
If the health check is not expected, then it uses the existing logic for a running task. I also expected it to be up, but currently it's unknown and I didn't know what the impact would be so i kept it as is.

Reply:
ok, so looks like `resultStatus` is set by `toPlatformHealthState` which never returns 'up', only 'unknown' at best. so that explains why there's no 'up' in this table.